src/pages/tasks/common/components/TaskDetails.tsx

```diff
@@ -41,26 +42,43 @@ export function TaskDetails(props: Props) {
 
   const { task, handleChange, errors } = props;
 
+  const handleChangeClient = (client: Client) => {
+  
+      handleChange('client_id', client.id);
+
+      if(!task?.id)
+        handleChange('rate', client?.settings?.default_task_rate ?? 0 );
+
+  }
+
+  const handleChangeProject = (project_id: string) => {
+
+    handleChange('project_id', project_id);
+    handleChange('client_id', '');
+
+  }
+
```

Comment:
I don't think we really need such methods, to be honest. Can we just cast multiple events out?

```tsx
onChange={(client) => {
    handleChange('client_id', client.id);
    handleChange('rate', ...)
}}
```

Reply:
as long as we can handle the conditional !task.id ... yep!